consensus/istanbul/backend/backend.go

```diff
@@ -992,3 +873,34 @@ func (sb *Backend) retrieveUncachedValidatorConnSet() (map[common.Address]bool,
 	logger.Trace("Returning validator conn set", "validatorsSet", validatorsSet)
 	return validatorsSet, nil
 }
+
+func (sb *Backend) sendForwardMsgToProxy(finalDestAddresses []common.Address, ethMsgCode uint64, payload []byte) error {
+	logger := sb.logger.New("func", "SendForwardMsg")
+	if sb.proxyNode.peer == nil {
+		logger.Warn("No connected proxy for sending a fwd message", "ethMsgCode", ethMsgCode, "finalDestAddreses", common.ConvertToStringSlice(finalDestAddresses))
+		return errNoProxyConnection
+	}
+
+	// Convert the message to a fwdMessage
+	fwdMessage := &istanbul.ForwardMessage{
+		Code:          ethMsgCode,
+		DestAddresses: finalDestAddresses,
+		Msg:           payload,
+	}
+	fwdMsgBytes, err := rlp.EncodeToBytes(fwdMessage)
+	if err != nil {
+		logger.Error("Failed to encode", "fwdMessage", fwdMessage)
+		return err
+	}
+
+	// Note that we are not signing message.  The message that is being wrapped is already signed.
+	msg := istanbul.Message{Code: istanbul.FwdMsg, Msg: fwdMsgBytes, Address: sb.Address()}
+	fwdMsgPayload, err := msg.Payload()
+	if err != nil {
+		return err
+	}
+
+	go sb.proxyNode.peer.Send(istanbul.FwdMsg, fwdMsgPayload)
```

Comment:
Ya, I totally agree that we should be consistent.  However, do you have strong opinions of doing this now?  My plan is to "phase" some of this refactoring as feature/bug fix PRs are being pushed.  Things on the roadmap are:

1). Create new subpackage for proxy (this will be part of the multiproxy PR).
2). Create new subpackage for announce.
3). Do more splitting up of the backend package and settle on a multithreaded vs singlethreaded messages.

Btw, historically, we have run into problems b/c of the "single threaded"-ness of our core package.  So I think that a significant amount of thought should go into #3.

Reply:
No strong opinion about this now and I agree this isn't a straightforward decision, so sounds good